### PR TITLE
Dynamic sitewide link rows

### LINKED ISSUE #567: Site wide links component doesnt support 8 items
**Describe the bug**
When you have 8 items in the site wide links component the last item is pulled out of the grid weirdly

**To Reproduce**

1. Add 8 items to component
2. See issue

**Expected behavior**
The items continue to fit into the grid columns

**Screenshots**
image

### PR DESCRIPTION
**Pull request by dr3**
Resolves #567

**Overall change:** Make sitewide link component rows dynamic

**Code changes:**

- Makes the number of rows used in sitewide links dynamic based on the number of children and desired number of columns at that break point. This is due to the desire to order the list like

| one | three | five | seven |
| two | four | six | eight |

not

| one | two | three | four |
| five | six | seven | eight |

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: packages/components/psammead-sitewide-links/src/List/index.jsx [hunk 1]
```diff
@@ -15,6 +15,11 @@ import {
 
 import Link from '../Link';
 
+// Gets the number of grid rows, taking into account the
+// first-child in the grid being seperate, on its own row.
```
Comment (DenisHdz) [inline packages/components/psammead-sitewide-links/src/List/index.jsx]:
```suggestion
// first-child in the grid being separate, on its own row.
```

File: packages/components/psammead-sitewide-links/src/List/index.jsx [hunk 2]
```diff
@@ -15,6 +15,11 @@ import {
 
 import Link from '../Link';
 
+// Gets the number of grid rows, taking into account the
+// first-child in the grid being separate, on its own row.
+const getRowCount = (children, columns) =>
```
Comment (DenisHdz) [inline packages/components/psammead-sitewide-links/src/List/index.jsx:22]: Nice one, I was doing something very similar yesterday 👌🏻

**Review by DenisHdz — DISMISSED** [review]
LGTM 👍

File: packages/components/psammead-sitewide-links/src/List/index.jsx [hunk 3]
```diff
@@ -15,6 +15,11 @@ import {
...
+  Math.ceil((children.length - 1) / columns) + 1;
```
Comment (andrew-nowak) [inline packages/components/psammead-sitewide-links/src/List/index.jsx]:
```suggestion
  Math.ceil((React.Children.count(children) - 1) / columns) + 1;
```
This handles single or zero components more gracefully than `.length`
https://reactjs.org/docs/react-api.html#reactchildrencount

**Review by andrew-nowak — CHANGES_REQUESTED** [review]

File: packages/components/psammead-sitewide-links/src/List/index.jsx (hunk 3 above)
Comment (dr3) [reply packages/components/psammead-sitewide-links/src/List/index.jsx]: Ohhh good spot, my bad :see_no_evil:

**Review by DenisHdz — DISMISSED** [review]

**Review by andrew-nowak — DISMISSED** [review]
:+1: very cool

**Review by DenisHdz — APPROVED** [review]
LGTM 👍

**Review by ghost — APPROVED** [review]
Great Work 👍

**Review by j-pendlebury — APPROVED** [review]

**Comment by jamesbrumpton** [comment]
LGTM.

**Pull request merged**